packages/wrangler/src/__tests__/config/findWranglerConfig.test.ts

```diff
@@ -0,0 +1,280 @@
+import path from "node:path";
+import { findWranglerConfig } from "../../config/config-helpers";
+import { mockConsoleMethods } from "../helpers/mock-console";
+import { normalizeString } from "../helpers/normalize";
+import { runInTempDir } from "../helpers/run-in-tmp";
+import { seed } from "../helpers/seed";
+
+describe("config findWranglerConfig()", () => {
+	runInTempDir();
+	const std = mockConsoleMethods();
+	const NO_LOGS = { debug: "", err: "", info: "", out: "", warn: "" };
+
+	describe("(useRedirectIfAvailable: false)", () => {
+		it.each(["toml", "json", "jsonc"])(
+			"should find the nearest wrangler.%s to the reference directory",
+			async (ext) => {
+				await seed({
+					[`wrangler.${ext}`]: "DUMMY",
+					[`foo/wrangler.${ext}`]: "DUMMY",
+					[`foo/bar/wrangler.${ext}`]: "DUMMY",
+					[`foo/bar/qux/holder.txt`]: "DUMMY",
+				});
+				expect(findWranglerConfig(".")).toEqual({
+					configPath: path.resolve(`wrangler.${ext}`),
+					userConfigPath: path.resolve(`wrangler.${ext}`),
+				});
+				expect(findWranglerConfig("./foo")).toEqual({
+					configPath: path.resolve(`foo/wrangler.${ext}`),
+					userConfigPath: path.resolve(`foo/wrangler.${ext}`),
+				});
+				expect(findWranglerConfig("./foo/bar")).toEqual({
+					configPath: path.resolve(`foo/bar/wrangler.${ext}`),
+					userConfigPath: path.resolve(`foo/bar/wrangler.${ext}`),
+				});
+				expect(findWranglerConfig("./foo/bar/qux")).toEqual({
+					configPath: path.resolve(`foo/bar/wrangler.${ext}`),
+					userConfigPath: path.resolve(`foo/bar/wrangler.${ext}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			}
+		);
+
+		describe.each([
+			["json", "jsonc"],
+			["json", "toml"],
+			["jsonc", "toml"],
+		])("should prefer the wrangler.%s over wrangler.%s", (ext1, ext2) => {
+			it("in the same directory", async () => {
+				await seed({
+					[`wrangler.${ext1}`]: "DUMMY",
+					[`wrangler.${ext2}`]: "DUMMY",
+				});
+				expect(findWranglerConfig(".")).toEqual({
+					configPath: path.resolve(`wrangler.${ext1}`),
+					userConfigPath: path.resolve(`wrangler.${ext1}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			});
+
+			it("in different directories", async () => {
+				await seed({
+					[`wrangler.${ext1}`]: "DUMMY",
+					[`foo/wrangler.${ext2}`]: "DUMMY",
+				});
+				expect(findWranglerConfig("./foo")).toEqual({
+					configPath: path.resolve(`wrangler.${ext1}`),
+					userConfigPath: path.resolve(`wrangler.${ext1}`),
+				});
+				expect(std).toEqual(NO_LOGS);
+			});
+		});
+
+		it("should return user config path even if a deploy config is found", async () => {
+			await seed({
+				[`wrangler.toml`]: "DUMMY",
+				[".wrangler/deploy/config.json"]: `{"configPath": "../../dist/wrangler.json" }`,
+				[`dist/wrangler.json`]: "DUMMY",
+			});
+			expect(
+				findWranglerConfig(".", { useRedirectIfAvailable: false })
```

Comment:
`{ useRedirectIfAvailable: false }` must be the default value because it is not specified for other calls to `findWranglerConfig` in this describe.

Should we be consistent in (not) using it. Another option would be to test both either explicitly`expect(
				findWranglerConfig({ useRedirectIfAvailable: false })).toEqual(findWranglerConfig())` or with a parametrized test (`[{useRedirectIfAvailable: false}, undefined]`)

Reply:
Removed.